### ISSUE TITLE
pgf update breaks tikz-3dplot?

### ISSUE THREAD

**Issue by ulilaube** [pgf-tikz/pgf#809]
Hi!
This morning I got an update from CTAN of the package `pgf`.

`[ 5/12, 00:03/00:11] update: pgf [11029k] (51817 -> 53187) ... done`

`tlmgr info pgf` and `\pgfversion` still report the version to be 3.1.5.
I am running the following MWE through `lualatex` 1.10.0 from TeX Live 2019 on my Linux box.

Yesterday I was working on a document containing drawings that uses the package `tikz-3dplot`. After the update this morning some drawings look different. The vectors specified via `\tdplotsetcoord` are not displayed correctly any more. They are all on top of the z-axis. That is vector `r` in the MWE. The vector `d` that has its coordinates specified manually as a triple is fine. Also the projections `Pxy`, ... of the vector onto the plane is no longer shown.

```latex
\documentclass{minimal}
\usepackage{tikz}
\usepackage{tikz-3dplot}
\begin{document}
\pgfversion
\tdplotsetmaincoords{70}{110}
\begin{tikzpicture}[scale=5,tdplot_main_coords]
\tdplotsetcoord{O}{0}{0}{0}
\tdplotsetcoord{P}{1.0}{60}{50}
\draw[->] (0,0,0) -- (1,0,0) node(xline)[anchor=south east]{$x$};
\draw[->] (0,0,0) -- (0,1,0) node(yline)[anchor=south west]{$y$};
\draw[->] (0,0,0) -- (0,0,1) node(zline)[anchor=south east]{$z$};
\draw[->] (O) -- (P) node[label={above left:$r$}] {};
\draw[->] (0,0,0) -- (1,1,1) node(zline)[anchor=south east]{$d$};
\draw[dashed] (O) -- (Pxy);
\draw[dashed] (P) -- (Pxy);
\draw[dashed] (Pxy) -- (Px);
\draw[dashed] (Pxy) -- (Py);
\end{tikzpicture}
\end{document}
```
After rolling the update back via

`tlmgr restore pgf 51667`

the MWE looks just fine. As Jeff Hein's `tikz-3dplot` is from 2012 I suppose something in `pgf` has changed in an incompatible way?

Happy Holidays!

**Comment by hmenke** [comment]
888f902c611c870523b3a9442b48001f1e160f69 fix for #785

**Comment by hmenke** [comment]
Hotfix is on its way.  Temporary workaround:
```tex
\documentclass{minimal}
\usepackage{tikz}
\usepackage{tikz-3dplot}

\makeatletter
\def\tikz@@coordinate@at@math#1{%
  \pgf@process{#1}%
  \edef\tikz@temp{(\the\pgf@x,\the\pgf@y)}%
  \expandafter\tikz@coordinate@caller\tikz@temp{}%
}%
\makeatother

\begin{document}
\pgfversion
\tdplotsetmaincoords{70}{110}
\begin{tikzpicture}[scale=5,tdplot_main_coords]
\tdplotsetcoord{O}{0}{0}{0}
\tdplotsetcoord{P}{1.0}{60}{50}
\draw[->] (0,0,0) -- (1,0,0) node(xline)[anchor=south east]{$x$};
\draw[->] (0,0,0) -- (0,1,0) node(yline)[anchor=south west]{$y$};
\draw[->] (0,0,0) -- (0,0,1) node(zline)[anchor=south east]{$z$};
\draw[->] (O) -- (P) node[label={above left:$r$}] {};
\draw[->] (0,0,0) -- (1,1,1) node(zline)[anchor=south east]{$d$};
\draw[dashed] (O) -- (Pxy);
\draw[dashed] (P) -- (Pxy);
\draw[dashed] (Pxy) -- (Px);
\draw[dashed] (Pxy) -- (Py);
\end{tikzpicture}
\end{document}
```
